src/HtsSystemContract.sol

```diff
@@ -194,6 +230,15 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events {
             require(msg.sender == HTS_ADDRESS, "getTokenInfo: unauthorized");
             _initTokenData();
             return abi.encode(22, _tokenInfo);
+        } else if (selector == this._update.selector) {
+            require(msg.data.length >= 124, "update: Not enough calldata");
+            require(msg.sender == HTS_ADDRESS, "update: unauthorized");
+            address from = address(bytes20(msg.data[40:60]));
+            address to = address(bytes20(msg.data[72:92]));
+            uint256 amount = uint256(bytes32(msg.data[92:124]));
+            _initTokenData();
```

Comment:
We shouldn't need this here. `_update` can also be called with transfers, where `getTokenInfo` shouldn't be needed. The main issue by having this here is that we may be triggering unnecessary requests to `getTokenInfo`.

Reply:
I know, yes, but still, we should remove this here. The reason is that by design is not be needed, so by removing it we may uncover some problems somewhere else.

Reply:
But `totalSupply+=amount` and `totalSupply-=amount` wouldn't work properly if we haven't initialized the token

Reply:
I see, yeah `totalSupply` needs to be initialized. Thanks for the clarification.